Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs

```diff
@@ -564,12 +587,35 @@ bookingCancelledReqHandler ValidatedBookingCancelledReq {..} = do
   unless (booking.status == DRB.CANCELLED) $ void $ QRB.updateStatus booking.id DRB.CANCELLED
   whenJust mbRide $ \ride -> void $ do
     unless (ride.status == DRide.CANCELLED) $ void $ QRide.updateStatus ride.id DRide.CANCELLED
+  fork "Cancellation Settlement" $ do
+    whenJust cancellationFee $ \fee -> do
+      riderConfig <- QRiderConfig.findByMerchantOperatingCityId booking.merchantOperatingCityId >>= fromMaybeM (InternalError "RiderConfig not found")
+      merchant <- CQM.findById booking.merchantId >>= fromMaybeM (MerchantNotFound booking.merchantId.getId)
+      person <- QP.findById booking.riderId >>= fromMaybeM (PersonNotFound booking.riderId.getId)
+      merchantCity <- CQMOC.findById booking.merchantOperatingCityId >>= fromMaybeM (MerchantOperatingCityNotFound booking.merchantOperatingCityId.getId)
+      mobileNumber <- mapM decrypt person.mobileNumber >>= fromMaybeM (PersonFieldNotPresent "mobileNumber")
+      case (riderConfig.settleCancellationFeeBeforeNextRide, mbRide, person.mobileCountryCode) of
+        (Just True, Just ride, Just countryCode) -> do
+          SPayment.makeCancellationPayment booking.merchantId booking.merchantOperatingCityId booking.riderId ride fee
+          --TODO: We can move this to stripe confirmation of payment
+          void $
+            CallBPPInternal.customerCancellationDuesSync
```

Comment:
What is this extra API for?

Reply:
To clear dues for the customer at BPP side.
We have customer cancellation dues in riderDetails table in BPP.
So this will deduct dues from it which customer has already paid to the driver.